sklearn/tree/export.py

```diff
@@ -373,8 +375,9 @@ def recurse(tree, node_id, criterion, parent=None, depth=0):
     return_string = False
     try:
         if out_file == 'tree.dot':
-            warnings.warn("out_file can be set to None from 0.18, "
-                          "this will be the default in 0.20.", DeprecationWarning)
+            warnings.warn("out_file can be set to None starting from 0.18. "
```

Comment:
Soo... this is a tricky one. You don't want to give the warning if a person explicitly provided the file.
Usually we would set the default to "None" (meaning not set) , test if the parameter was none, raise the warning, and do the old behavior. Then, in the future, we would replace the None by the new parameter.
The problem here is that if someone wanted the new behavior now, they would set the parameter to None, and get a deprecation warning.

There's two solutions: use another marker value to show that the user didn't set the option (we could introduce a ``DeprecatedValue`` class and test for that) or we change the new value to something other than ``None``, say ``"none"`` (the string "none"). I think we usually strive for simple, so maybe the second option is better?